modules/apps/notification/notification-test/src/testIntegration/java/com/liferay/notification/internal/type/test/EmailNotificationTypeTest.java

```diff
@@ -519,15 +522,49 @@ public void testSendNotificationWithRegularRoles() throws Exception {
 					RandomTestUtil.randomString(),
 					NotificationConstants.TYPE_EMAIL, Collections.emptyList()));
 
+		ObjectDefinition objectDefinition =
+			_addAndPublishCustomObjectDefinition(false);
+
+		_addObjectAction(
+			objectDefinition.getObjectDefinitionId(),
+			ObjectActionTriggerConstants.KEY_ON_AFTER_ADD,
+			notificationTemplate.getNotificationTemplateId());
+
 		_testSendNotificationWithRoles(
-			null, StringPool.BLANK, 0, null, notificationTemplate);
+			null, StringPool.BLANK, 0, null,
+			ObjectActionTriggerConstants.KEY_ON_AFTER_ADD, objectDefinition);
 
 		_roleLocalService.addUserRole(user1.getUserId(), role1.getRoleId());
-		_roleLocalService.addUserRole(user2.getUserId(), role2.getRoleId());
+
+		User user = userLocalService.addUser(
```

Comment:
@pedro-oliveira446  Why you're not using the existing users here (user1 and user2)?

Reply:
I not used user2 because the base setup assigns the user permission checker to user2. As a result, the object entry is saved with user2's userId, giving them owner permissions. However, I made changes to this in the last commit.